### ISSUE TITLE
Weird Artwork Behaviour

### ISSUE THREAD

**Issue by vcdx71** [music-assistant/hass-music-assistant#2045]
### What version of Music Assistant has the issue?

2.0.0B118

### What version of the Home Assistant Integration have you got installed?

2024.3.4

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Using the standard Home Assistant Media Player Card.

Choosing and playing a song displays the artwork as expected on the card. However, when the next song in the album plays the artwork is gone. If I manually start playing the same song the artwork displays correctly. 

If an album is played artwork never shows up.

If an artist is played artwork for every song displays correctly. 

The behavior was the same in B117. Can't comment on previous version as I switched from Plex to Local Files, though I did not experience this issue with the Plex provider. 

### How to reproduce

Play songs, albums, and artists either through Music Assistant interface for via the Home Assistant media browser on the media player card and observe the above artwork weirdness. 

### Music Providers

Only Local Files

### Player Providers

ChromeCast, AirPlay, and DNLA.

### Full log output

[ma.log](https://github.com/music-assistant/hass-music-assistant/files/14716301/ma.log)

### Additional information

Artwork is embedded in each file and also in a folder.jpg in each album directory. 

### What version of Home Assistant Core are your running

2024.3.1

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by marcelveldt** [comment]
Thanks for the very clear bug report, I'll look into it!

**Comment by S1M8N** [comment]
Same in playlist section

**Comment by marcelveldt** [comment]
this should be fixed with MA server 2.20.0 b146 + HA integration 2024.5.0 which are just released.

**Comment by vcdx71** [comment]
Thanks looks like the media player is displaying artwork correctly now.

I am though now seeing no artwork for artists or albums.
![image](https://github.com/music-assistant/hass-music-assistant/assets/16821128/98fee620-5133-4089-beea-2fb345d9389c)
![image](https://github.com/music-assistant/hass-music-assistant/assets/16821128/4356105c-b06c-4c67-9de9-a1d16d2d3c58)

**Comment by marcelveldt** [comment]
Does the artwork show up in the MA interface ? Are these local items or streaming providers ?

**Comment by vcdx71** [comment]
They are local and they show perfectly in the MA interface.

**Comment by marcelveldt** [comment]
I finally had some time to dig into this issue. 
Is it true that the images only do not work if you are accessing HA on its remote URL ?

**Comment by OzGav** [comment]
I can recreate this and with the local IP URL (http://192.168.1.166:8123/) I don't see any album thumbs. Artist thumbs are OK (minus the disappearing ones subject to a separate report)

**Comment by vcdx71** [comment]
> I finally had some time to dig into this issue. 
> Is it true that the images only do not work if you are accessing HA on its remote URL ?

Hadn't tested it that way, but yes you're correct. Artwork loads when accessing via IP, I usually use a cloudflare tunnel.

**Comment by marcelveldt** [comment]
> I can recreate this and with the local IP URL (http://192.168.1.166:8123/) I don't see any album thumbs. Artist thumbs are OK (minus the disappearing ones subject to a separate report)

Yeah, that is what I suspected. Looks like I need to do some special magic handling to translate the urls.

**Comment by marcelveldt** [comment]
Unfortunately this is not possible without

a) a lot of work - its not as straightforward as expected
b) impact on speed

As the HA mediabrowser is not our main priority (its lacking lots of features anyways) I'm not going to invest hours of work in this now while other more urgent issues are open.

I'm going to propose a change in the HA api to make this easier to work with.

I will leave this one open here to pickup on a rainy day.

For reference: Local artwork (as in: on your local disk/network) will NOT show up in the HA MediaBrowser if you are connecting to HA using an external URL.

**Comment by vcdx71** [comment]
Keeping from going stale

**Comment by Robert-MacWha** [comment]
The issue is still present.  Some basic diagnostics I tried:

1. When I had a "Filesystem (remote share)" provider, artists would be visible in the media browser, requesting images from "r2.theaudiodb.com".  Album art was not visible, and the requests for the album images were made to "172.30.32.1:8095" (_the address on which the "music_assistant.webserver" server is running with HAOS_)
2. When I switched to a Jellyfin provider, artists became no longer available.  Now all requests are made to "172.30.32.1:8095".  It seems like Music Assistant is downloading the artwork from Jellyfin and putting it into local storage(?) because when I change an artist's artwork in Jellyfin it's reflected in Music Assistant.  

All the above was from a raspberry pi 3b+ running HAOS, CORE=2024.11.3, Music Assistant=2024.11.5, connected to Home Assistant over LAN via its IP address (192.168.2.163:8123).

Digging around in the code - `def get_image_url` in "https://github.com/music-assistant/client" is probably the method to blame.  For local images (_which I'm fairly sure we're dealing with_) it uses the `self.server_info.base_url` which presumably equals "172.30.32.1:8095".

I'm having trouble figuring out how to setup the full project right now, but it seems like we could just change the method to use the `hassio_ingress` API and fetch the image.  Use the same URLs as Music Assistant's dashboard. 

Let me know if I'm barking up the wrong tree here.  I don't see the performance / complexity issues @marcelveldt mentioned above, but I could definitely be wrong.

**Comment by Akruidenberg** [comment]
Same here. MusicAssistant 2.3.4 and HA 2024.12.5

**Comment by chatziko** [comment]
I had the same issue. A workaround is to use an https reverse proxy. Then configure the https URL in MA under `/#/settings/editcore/webserver`. Now local artowork shows fine in HA.

**Comment by vcdx71** [comment]
> I had the same issue. A workaround is to use an https reverse proxy. Then configure the https URL in MA under `/#/settings/editcore/webserver`. Now local artowork shows fine in HA.

Perfect, not sure why I never thought of that, but can confirm this works!

**Comment by balgerion** [comment]
Does this workaround still work? I don't see any option to configure HTTPS at that link


![Image](https://github.com/user-attachments/assets/715781b9-5386-49f9-94f3-863d9678fffb)
